examples/stm32h723/src/bin/spdifrx.rs

```diff
@@ -0,0 +1,160 @@
+
+bind_interrupts!(struct Irqs {
+    SPDIF_RX => spdifrx::GlobalInterruptHandler<peripherals::SPDIFRX1>;
+});
+
+const CHANNEL_COUNT: usize = 2;
+const BLOCK_LENGTH: usize = 64;
+const HALF_DMA_BUFFER_LENGTH: usize = BLOCK_LENGTH * CHANNEL_COUNT;
+const DMA_BUFFER_LENGTH: usize = HALF_DMA_BUFFER_LENGTH * 2; //  2 half-blocks
+
+// DMA buffers must be in special regions. Refer https://embassy.dev/book/#_stm32_bdma_only_working_out_of_some_ram_regions
+#[link_section = ".sram1"]
+static mut SPDIFRX_BUFFER: GroundedArrayCell<u32, DMA_BUFFER_LENGTH> = GroundedArrayCell::uninit();
+
+#[link_section = ".sram4"]
+static mut SAI_BUFFER: GroundedArrayCell<u32, DMA_BUFFER_LENGTH> = GroundedArrayCell::uninit();
+
+#[embassy_executor::main]
+async fn main(_spawner: Spawner) {
+    let mut peripheral_config = embassy_stm32::Config::default();
+    {
+        use embassy_stm32::rcc::*;
+        peripheral_config.rcc.hsi = Some(HSIPrescaler::DIV1);
+        peripheral_config.rcc.pll1 = Some(Pll {
+            source: PllSource::HSI,
+            prediv: PllPreDiv::DIV16,
+            mul: PllMul::MUL200,
+            divp: Some(PllDiv::DIV2), // 400 MHz
+            divq: Some(PllDiv::DIV2),
+            divr: Some(PllDiv::DIV2),
+        });
+        peripheral_config.rcc.sys = Sysclk::PLL1_P;
+        peripheral_config.rcc.ahb_pre = AHBPrescaler::DIV2;
+        peripheral_config.rcc.apb1_pre = APBPrescaler::DIV2;
+        peripheral_config.rcc.apb2_pre = APBPrescaler::DIV2;
+        peripheral_config.rcc.apb3_pre = APBPrescaler::DIV2;
+        peripheral_config.rcc.apb4_pre = APBPrescaler::DIV2;
+
+        peripheral_config.rcc.mux.spdifrxsel = mux::Spdifrxsel::PLL1_Q;
+    }
+    let mut p = embassy_stm32::init(peripheral_config);
+
+    info!("SPDIFRX to SAI4 bridge");
+
+    // Use SPDIFRX clock for SAI.
+    // This ensures equal rates of sample production and consumption.
+    let clk_source = embassy_stm32::pac::rcc::vals::Saiasel::_RESERVED_5;
+    embassy_stm32::pac::RCC.d3ccipr().modify(|w| {
+        w.set_sai4asel(clk_source);
+    });
+
+    let sai_buffer: &mut [u32] = unsafe {
+        SAI_BUFFER.initialize_all_copied(0);
+        let (ptr, len) = SAI_BUFFER.get_ptr_len();
+        core::slice::from_raw_parts_mut(ptr, len)
+    };
+
+    let spdifrx_buffer: &mut [u32] = unsafe {
+        SPDIFRX_BUFFER.initialize_all_copied(0);
+        let (ptr, len) = SPDIFRX_BUFFER.get_ptr_len();
+        core::slice::from_raw_parts_mut(ptr, len)
+    };
+
+    let mut spdif_receiver = new_spdif_receiver(&mut p.SPDIFRX1, &mut p.PD7, &mut p.DMA2_CH7, spdifrx_buffer);
+    let mut sai_transmitter = new_sai_transmitter(
+        &mut p.SAI4,
+        &mut p.PD13,
+        &mut p.PC1,
+        &mut p.PD12,
+        &mut p.BDMA_CH0,
+        sai_buffer,
+    );
+
+    spdif_receiver.start();
+
+    loop {
+        let mut buf = [0u32; HALF_DMA_BUFFER_LENGTH];
+
+        match spdif_receiver.read_data(&mut buf).await {
```

Comment:
As of yesterday's discoveries, should there now be a `select` with `wait_write_error()` here?

Reply:
You can have a look, I changed it accordingly.
It can recover cleanly from a loss of the S/PDIF signal.